### ISSUE TITLE
Use system-cxx-std-lib virtual package where available

### ISSUE THREAD

**Issue by Bodigrim** [haskell/double-conversion#33]
Similar to https://github.com/haskell/text/pull/455.

**Comment by Haskell-mouse** [comment]
You also suggested to remove version identifier from `stdc++`. I've checked that this identifier was introduced a long time ago 
in the following pull request: #13 due to a [GHC bug](https://gitlab.haskell.org/ghc/ghc/-/issues/5289). 
The bug is closed, I've checked on windows that linking against `stdc++` without version identifier works well at leas with GHC 8.6.5. 
I'll probably make this version as the oldest for linking against `stdc++` instead of `stdc++-6`.

**Comment by Haskell-mouse** [comment]
Fixed in 5f3941175978ebf160f2d6b8fa8e13e16c3b6ed8 and in 9b919c444882a5e665cd59693a89f3e291bba210

**Comment by Bodigrim** [comment]
@Haskell-mouse `cabal build` says
```
Warning: double-conversion.cabal:86:3: invalid subsection "else"
Warning: double-conversion.cabal:77:3: invalid subsection "elif"
Warning: double-conversion.cabal:75:3: invalid subsection "elif". You should
set cabal-version: 2.2 or larger to use elif-conditionals.
```

**Comment by Haskell-mouse** [comment]
Oh, thanks. I'll set the version of cabal.
It is weird that I haven't gotten these warning from Cabal.

**Comment by Haskell-mouse** [comment]
> @Haskell-mouse `cabal build` says
> 
> ```
> Warning: double-conversion.cabal:86:3: invalid subsection "else"
> Warning: double-conversion.cabal:77:3: invalid subsection "elif"
> Warning: double-conversion.cabal:75:3: invalid subsection "elif". You should
> set cabal-version: 2.2 or larger to use elif-conditionals.
> ```
@Bodigrim , fixed this.

**Comment by Bodigrim** [comment]
Thanks, it works now.

**Comment by tfausak** [comment]
Would you mind making a new release to Hackage with these changes?

**Comment by tfausak** [comment]
For posterity, this was released with 0d8ea3db291f5cdece4e60508b0f4ce3ee70288e in [version 2.0.4.2](https://hackage.haskell.org/package/double-conversion-2.0.4.2).
